dbt/models/iasworld/schema/iasworld.comdat.yml

```diff
@@ -101,21 +101,30 @@ sources:
           - name: chgrsn
             description: '{{ doc("column_chgrsn") }}'
             tests:
-              - expression_is_true:
-                  name: iasworld_comdat_chgrsn_eq_5_mktadj_not_null
-                  expression: |
-                    != '5' OR mktadj IS NOT NULL
+              - accepted_values:
+                  name: iasworld_comdat_chgrsn_eq_5_or_5b_when_external_occpct_or_mktadj_not_null
+                  values:
+                    - '5'
+                    - 5B
                   additional_select_columns:
                     - parid
                     - taxyr
                     - card
                     - who
                     - wen
+                    - external_occpct
                     - mktadj
-                  config: *unique-conditions
+                  config:
+                    where: |
+                      CAST(taxyr AS int) BETWEEN {{ var('test_qc_year_start') }} AND {{ var('test_qc_year_end') }}
+                      AND cur = 'Y'
+                      AND deactivat IS NULL
+                      AND (external_occpct IS NOT NULL OR mktadj IS NOT NULL)
                   meta:
-                    category: incorrect_values
-                    description: chgrsn should only be 5 if mktadj is not null
+                    description: >
+                      chgrsn (Reason for Change) should be 5 or 5B
+                      if external_occpct (Occupancy % [current]) or
+                      mktadj (Occupancy % [deprecated]) is not null
```

Comment:
Note that this test logic is not as thorough as it could be; we've been told that the presence of `mktadj` or `external_occpct` should correspond to a specific timeline (see PR description), but so far these tests do not capture that time dimension. Do you think it's worth refactoring the tests to specify that `mktadj` should be the populated field pre-2023, and `external_occpct` should be the populated field post-2024?

Reply:
No, if we can't correct historical data then there's not really a point to worrying about a time dimension for any tests IMO.

Reply:
That makes sense @dfsnow. I wonder if perhaps we should just test `external_occpct` then, and leave `mktadj` out of it?

Reply:
I think it's fine to test both, on the off chance that someone accidentally uses `mktadj`.